TestBallastMemory - Intermittent test failure

The test TestBallastMemory failed during PR #3128. Unfortunately, the test does not specify what's the memory consumption, only the limit.

```
=== RUN   TestBallastMemory
make: *** [Makefile:60: testbed-loadtest] Error 1
2021/05/18 21:40:40 Starting Agent (/home/runner/work/opentelemetry-collector/opentelemetry-collector/bin/otelcol_linux_amd64)
2021/05/18 21:40:40 Writing Agent log to /home/runner/work/opentelemetry-collector/opentelemetry-collector/testbed/tests/results/TestBallastMemory/agent.log
2021/05/18 21:40:40 Agent running, pid=7649
2021/05/18 21:40:40 Stopped generator. Sent:         0 items
2021/05/18 21:40:40 Gracefully terminating Agent pid=7649, sending SIGTEM...
2021/05/18 21:40:40 Stopping process monitor.
2021/05/18 21:40:40 Agent process stopped, exit code=0
2021/05/18 21:40:40 Starting Agent (/home/runner/work/opentelemetry-collector/opentelemetry-collector/bin/otelcol_linux_amd64)
2021/05/18 21:40:40 Writing Agent log to /home/runner/work/opentelemetry-collector/opentelemetry-collector/testbed/tests/results/TestBallastMemory/agent.log
2021/05/18 21:40:40 Agent running, pid=7658
    e2e_test.go:88: 
        	Error Trace:	e2e_test.go:88
        	Error:      	Should be true
        	Test:       	TestBallastMemory
        	Messages:   	RSS must be less than or equal to 70
2021/05/18 21:40:40 Stopped generator. Sent:         0 items
2021/05/18 21:40:40 Gracefully terminating Agent pid=7658, sending SIGTEM...
2021/05/18 21:40:40 Stopping process monitor.
2021/05/18 21:40:40 Agent process stopped, exit code=0
2021/05/18 21:40:40 Starting Agent (/home/runner/work/opentelemetry-collector/opentelemetry-collector/bin/otelcol_linux_amd64)
2021/05/18 21:40:40 Writing Agent log to /home/runner/work/opentelemetry-collector/opentelemetry-collector/testbed/tests/results/TestBallastMemory/agent.log
2021/05/18 21:40:40 Agent running, pid=7666
2021/05/18 21:40:40 Stopped generator. Sent:         0 items
2021/05/18 21:40:40 Gracefully terminating Agent pid=7666, sending SIGTEM...
2021/05/18 21:40:40 Stopping process monitor.
2021/05/18 21:40:40 Agent process stopped, exit code=0
--- FAIL: TestBallastMemory (0.24s)
```

--- a/testbed/tests/e2e_test.go
+++ b/testbed/tests/e2e_test.go
@@ -84,7 +84,11 @@ func TestBallastMemory(t *testing.T) {
 			return vms > test.ballastSize
 		}, time.Second*2, "VMS must be greater than %d", test.ballastSize)
 
-		assert.LessOrEqual(t, rss, test.maxRSS)
+		// https://github.com/open-telemetry/opentelemetry-collector/issues/3233
+		// given that the maxRSS isn't an absolute maximum and that the actual maximum might be a bit off,
+		// we give some room here instead of failing when the memory usage isn't that much higher than the max
+		lenientMax := 1.1 * float32(test.maxRSS)
+		assert.LessOrEqual(t, rss, lenientMax)
 		tc.Stop()
 	}
 }

